gestalt-module/src/main/java/org/terasology/gestalt/naming/Name.java

```diff
@@ -39,62 +39,81 @@
      */
     public static final Name EMPTY = new Name("");
 
+    private final String normalised;
     private final String originalName;
-    private final String normalisedName;
 
     public Name(String name) {
         Preconditions.checkNotNull(name);
         this.originalName = name;
-        this.normalisedName = name.toLowerCase(Locale.ENGLISH);
+        this.normalised = name.toLowerCase(Locale.ENGLISH);
     }
 
     /**
      * @return Whether this name is empty (equivalent to an empty string)
      */
     public boolean isEmpty() {
-        return normalisedName.isEmpty();
+        return normalised.isEmpty();
     }
 
     /**
      * @return The Name in lowercase consistent with Name equality (so two names that are equal will have the same lowercase)
+     * @deprecated This is scheduled for removal in upcoming versions.
+     * Use {@code toString} or {@code displayName} instead.
+     * Note that a Name should not be transformed to a String for further processing.
      */
+    @Deprecated
     public String toLowerCase() {
-        return normalisedName;
+        return normalised;
     }
 
     /**
      * @return The Name in uppercase consistent with Name equality (so two names that are equal will have the same uppercase)
+     * @deprecated This is scheduled for removal in upcoming versions.
+     * Use {@code toString} or {@code displayName} instead.
+     * Note that a Name should not be transformed to a String for further processing.
      */
+    @Deprecated
     public String toUpperCase() {
         return originalName.toUpperCase(Locale.ENGLISH);
     }
 
     @Override
     public int compareTo(Name o) {
-        return normalisedName.compareTo(o.normalisedName);
+        return normalised.compareTo(o.normalised);
     }
 
+    /**
+     * normalises the string and compares it to the normalised version used for @{@linkplain Name}
+     * @param other string to compare against
+     * @return string comparision with normalised version 0 if match and unmatched with anything else
+     */
+    public int compareTo(String other) {
+        return normalised.compareTo(other.toLowerCase(Locale.ENGLISH));
+    }
+
+
     @Override
     public boolean equals(Object obj) {
         if (obj == this) {
             return true;
         }
         if (obj instanceof Name) {
             Name other = (Name) obj;
-            return normalisedName.equals(other.normalisedName);
+            return normalised.equals(other.normalised);
         }
         return false;
     }
 
     @Override
     public int hashCode() {
-        return normalisedName.hashCode();
+        return normalised.hashCode();
     }
 
+    /**
+     * @return The Name as string consistent with Name equality (so two names that are equal will have the same {@code toString})
```

Comment:
this no longer holds with the change reverted 😟 Now, we should rather put a disclaimer that it is only intended for display purposes.
```suggestion
     * @return The Name as string for display purposes
```